### ISSUE TITLE
Angular CLI V15 + TS Target ES2022 - No longer tree shaking unused services correctly

### ISSUE THREAD

**Issue by Coly010** [angular/angular-cli#24656]
### Command

build

### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

Angular v14

### Description

Unused Services are not being tree-shaken correctly from the bundle output with

`"target": "es2022"` set in the `tsconfig.json` and Angular v15 installed.

This target is set by the Angular v15 migrations!

### Minimal Reproduction

I've created a repository.

1. Clone https://github.com/Coly010/ng-tree-shaking

## The Problem

With Angular V15, when`"target": "es2022` is set in `tsconfig.json`, `UnusedService` a service that is not used in the application, is not tree shaken from the bundle.

This can be reproduced by doing the following"

1. Checkout `angular-v15-es2022`
2. Run `npm install`
3. Run `npm run build`
4. Run `npm run check-tree-shaking`

You'll see a line output stating that `grep` managed to find the `UnusedService` in the bundle.

### This is a regression

To prove this is a regression, you can do the following:

1. Checkout `angular-v14-es2022`
2. Run `npm install`
3. Run `npm run build`
4. Run `npm run check-tree-shaking`

You'll not see a line output stating that `grep` managed to find the `UnusedService` in the bundle.

### Futhermore

If you set the `"target": "es2020` in `tsconfig.json` with Angular v15, the service is tree shaken correctly.

1. Checkout `angular-v15-es2020`
2. Run `npm install`
3. Run `npm run build`
4. Run `npm run check-tree-shaking`

You'll not see a line output stating that `grep` managed to find the `UnusedService` in the bundle.

### Exception or Error

```text
Unused Service is being included in the bundle when it should be tree-shaken.
```

### Your Environment

```text
Angular: 15.1.3
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1501.4
@angular-devkit/build-angular   15.1.4
@angular-devkit/core            15.1.4
@angular-devkit/schematics      15.1.4
@angular/cli                    15.1.4
@schematics/angular             15.1.4
rxjs                            7.5.7
typescript                      4.9.5

```
```


### Anything else relevant?

_No response_

**Comment by alan-agius4** [comment]
This seems to have been caused by the introduction of `@babel/plugin-proposal-class-properties`.  in  https://github.com/angular/angular-cli/pull/24357

**Comment by devversion** [comment]
yeah, looks like it's because with ES2022 the static fields are preserved by TS (given no `useDefineForClassFields=false`). The output is then processed by Babel to workaround the Safari v15 bug.

https://babeljs.io/repl/#?browsers=last%202%20safari%20version&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=MYGwhgzhAECC0G8BQ1pmgXmgFwE4FcBTAbhWgmzGwEthoAPTHAkpAXyA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.20.15&externalPlugins=%40babel%2Fplugin-proposal-class-properties%407.18.6&assumptions=%7B%7D

```js
class A {
  constructor() {
    _defineProperty(this, "a", true);
  }
}
_defineProperty(A, "x", true);
```

Angular CLI previously only ever saw assignments (`A.x = true`), and no `_defineProperty` call. The CLI https://github.com/angular/angular-cli/blob/bf3be56db8b3a1ef8a0e02dfc5de914651d8ec93/packages/angular_devkit/build_angular/src/babel/plugins/adjust-static-class-members.ts Babel plugin does not understand this approach and likely needs to be updated to support it.
